imap_processing/codice/codice_l1a.py

```diff
@@ -111,16 +111,32 @@ def create_science_dataset(self, start_time: np.datetime64) -> xr.Dataset:
             attrs=cdf_attrs.get_variable_attributes("energy_attrs"),
         )
 
+        # Define labels
+        energy_label = xr.DataArray(
+            energy_steps.values.astype(str),
+            name="energy_label",
+            dims=["energy_label"],
+            attrs=cdf_attrs.get_variable_attributes("energy_label"),
+        )
+
         # Create the dataset to hold the data variables
         dataset = xr.Dataset(
-            coords={"epoch": epoch, "energy": energy_steps},
+            coords={
+                "epoch": epoch,
+                "energy": energy_steps,
+                "energy_label": energy_label,
+            },
             attrs=cdf_attrs.get_global_attributes(self.dataset_name),
         )
 
-        # Create a data variable for each species
+        # Create a data variable for each counter
         for variable_data, variable_name in zip(self.data, self.variable_names):
-            variable_data_arr = np.array(list(variable_data), dtype=int).reshape(
-                -1, self.num_energy_steps
+            # TODO: Currently, cdflib doesn't properly write/read CDF files that
```

Comment:
It would be good to broadcast this issue a bit more widely. There are several/many of us with products that have a single epoch value that should be made aware of the issue. I'd like to know some more details about how the issue manifests.

Reply:
Good suggestion, I will mention it in slack